File: fun-with-numbers/java_fun_with_numbers/src/main/java/com/pfc/allPrimes.java
package com.pfc;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class allPrimes {
    public static void main(String[] args) {
        start(999999999);

    }

    private static Result processOdds(int n, List<Integer> primes){
        double squared = Math.sqrt(n);
        squared = Math.floor(squared);
        Result result = new Result();
        int range = (int) squared;
        for (int i = 3; i <= range; i +=2) {
            result = processNumber(n, i, primes);
            n = result.getNumber();
        }
        if(n > 2){
            result.addPrime(n);
        }
        return result;
    }

    private static Result processNumber(int n, int i, List<Integer> primes){
        boolean isRunning = true;
        while(isRunning){
            if(n % i == 0){
                primes.add(i);
                n = n/i;
            }else{
                isRunning = false;
            }
        }
        Result result = new Result();
        result.setNumber(n);
        result.setPrimes(primes);
        return result;
    }

    private static void start(int number){
        List<Integer> primes = new ArrayList<>();
        Result result = processNumber(number, 2, primes);
        result = processOdds(result.getNumber(), result.getPrimes());
        System.out.println(result.getPrimeSet());
        System.out.println(result.getPrimes());
    }
    private static class Result{
        private int number;
        private List<Integer> primes;


        public int getNumber() {
            return number;
        }

        public void setNumber(int number) {
            this.number = number;
        }

        public List<Integer> getPrimes() {
            return primes;
        }

        public void setPrimes(List<Integer> primes) {
            this.primes = primes;
        }

        public void addPrime(int i) {
            this.primes.add(i);
        }

        public HashSet<Integer> getPrimeSet() {
            return new HashSet<Integer>(this.getPrimes());
        }
    }
}

File: fun-with-numbers/all_primes.py
import math

def process_numbers(n, i, primes):
    running = True
    while running:
        if n % i == 0:
            primes.append(i)
            n = n/i
        else:
            running = False
    return {
        "primes": primes,
        "number": n
    }

def process_odds(n, primes):
    x = int(round(math.sqrt(n), 0))
    for i in range(3,x,2):
        result = process_numbers(n, i, primes)
        n = int(result["number"])
    if n > 2:
        result["primes"].append(n)
    return result


def start(number):
    result = process_numbers(number, 2, [])
    result = process_odds(result["number"], result["primes"])
    print(result["primes"])
    print(set(result["primes"]))



if __name__ == "__main__":
    start(123456543457)